settings/resource_notification_destination.go

```diff
@@ -0,0 +1,177 @@
+package settings
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func setStruct(s *settings.NotificationDestination, readND *settings.NotificationDestination) {
+	switch readND.DestinationType {
+	case settings.DestinationTypeSlack:
+		readND.Config.Slack.Url = s.Config.Slack.Url
+	case settings.DestinationTypePagerduty:
+		readND.Config.Pagerduty.IntegrationKey = s.Config.Pagerduty.IntegrationKey
+	case settings.DestinationTypeMicrosoftTeams:
+		readND.Config.MicrosoftTeams.Url = s.Config.MicrosoftTeams.Url
+	case settings.DestinationTypeWebhook:
+		if readND.Config.GenericWebhook.UrlSet {
+			readND.Config.GenericWebhook.Url = s.Config.GenericWebhook.Url
+		}
+		if readND.Config.GenericWebhook.PasswordSet {
+			readND.Config.GenericWebhook.Password = s.Config.GenericWebhook.Password
+		}
+		if readND.Config.GenericWebhook.UsernameSet {
+			readND.Config.GenericWebhook.Username = s.Config.GenericWebhook.Username
+		}
+	}
+}
+
+func Create(ctx context.Context, d *schema.ResourceData, w *databricks.WorkspaceClient) error {
+	var newNDrequest settings.CreateNotificationDestinationRequest
+	common.DataToStructPointer(d, ndSchema, &newNDrequest)
+	createdND, err := w.NotificationDestinations.Create(ctx, newNDrequest)
+	if err != nil {
+		return err
+	}
+	d.SetId(createdND.Id)
+	d.Set("destination_type", createdND.DestinationType)
+	return nil
+}
+
+func Read(ctx context.Context, d *schema.ResourceData, w *databricks.WorkspaceClient) error {
+	var tempND settings.NotificationDestination
+	common.DataToStructPointer(d, ndSchema, &tempND)
+
+	readND, err := w.NotificationDestinations.Get(ctx, settings.GetNotificationDestinationRequest{
+		Id: d.Id(),
+	})
+	if err != nil {
+		return err
+	}
+	setStruct(&tempND, readND)
+	return common.StructToData(readND, ndSchema, d)
+}
+
+func detectConfigTypeChange(d *schema.ResourceData) bool {
```

Comment:
I'm not sure that we need this function. Also, it doesn't cover changes in `display_name`

Reply:
This function only covers the fields which would require this resource to be deleted and recreated. The `display_name` can be change using the update call